src/main/java/ru/skypro/homework/service/MyUserDetailsService.java

```diff
@@ -20,6 +26,63 @@ public UserDetails loadUserByUsername(String username) {
         if (user == null) {
             throw new UsernameNotFoundException(username);
         }
-        return new MyUserPrincipal(user);
+        return user;
+    }
+
+    @Override
+    public void createUser(UserDetails user) {
+//нужна реализация
+        userRepository.save((User) user);
```

Comment:
Так работать не будет. Нужно создавать нового юзера.
```suggestion
        User u = new User();
        u.setUserName(user.getUsername());
        u.setPassword(user.getPassword());
        u.setRole(Role.valueOf(user.getAuthorities().iterator().next().getAuthority().substring(5)));
        userRepository.save(u);
```